Issue 8 new readme and sample

dwk2 opened the pull request:
Completely new readme and sample file; improved documentation.

dwk2 committed:
Completely new readme and sample file; improved documentation.

dwk2 committed:
minor final edit

updated domain name in sample

dwk2 committed:
minor text update

added link back to repo for full instructions on readme file

patdunlavey commented on README.md:
````diff
@@ -1,122 +1,129 @@
+* ```drush help sitespinner```            [documentation for sitespinner create]
+* ```drush help sitespinner-delete```     [documentation for sitespinner delete]
+* ```drush cc drush```                    [clear drush cache] [shortcut: "cc" = "cache-clear"]
+* ```drush topic docs-aliases```          [helpful information on writing and installing drush site aliases]
````
This is a duplicate.

patdunlavey commented on README.md:
```diff
@@ -125,35 +132,24 @@ files.
-Force the script to continue when errors are encountered. E.g.:
-You have tried to install a site but your failed halfway due to permissions.
-You can then change the permissions on your server, and force the script to execute.
+* **Chris Warren** - *Initial version of Sitespinner in PHP*
```
This is not correct. Sitespinner was originally written as bash scripts by Andy Cavenaugh, then at Common Media. I rewrote it as a drush command. Chris did not do any coding, as I recall, but of course he developed the concept and oversaw development of both versions.

patdunlavey commented on examples/sitespinner-sample.aliases.drushrc.php:
```diff
@@ -0,0 +1,209 @@
+	$DRUPAL_WEB_ROOT_PATH = "/var/www/html";                    # "root" is where the web server's index.php file is located, not the multi-site root; omit trailing slash "/"
+	$SECURITY_MODE        = "https://";                         # "https://" or "http://"
+	#--- MySQL authentication (consider granting all privileges for this db user: absolutely requires CREATE, SELECT, INSERT, DELETE, DROP) ---#
+	$MYSQL_HOST_NAME = "subdomain-webdb.domain.edu";            # full host name (include subdomain, if exists) of server running MySQL (mysql.domain.edu)
```
You really should provide two mysql users, one for the site creator, and the other for the drupal user. The user that runs the site should not have the ability to create and destroy databases.

patdunlavey commented:
Hi David, Excellent work, thank you so much!! 

I found a few things that I think need to be fixed. See my comments.

dwk2 committed:
readme corrections

updated original author info and removed duplicate line

dwk2 committed:
added TODO item

added TODO to improve workflow security in regards to mysql db users and their permissions

dwk2 commented:
Hi Pat,
Thank you for your careful read. I've completed your suggested corrections:
- removed duplicate line
- corrected author information
- added a TODO item noting: "Consider providing two mysql users: one for
the site creator, and the other for the drupal user. The user that runs the
site should not have the ability to create and destroy databases."

The final one I've put on my own TODO list to complete later.

If you think this is ready, please merge.

Thank you,
David

On Wed, Jun 14, 2017 at 7:13 AM, Pat Dunlavey <notifications@github.com>
wrote:

> Hi David, Excellent work, thank you so much!!
>
> I found a few things that I think need to be fixed. See my comments.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Islandora-Collaboration-Group/drush-sitespinner/pull/9#issuecomment-308400331>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADME34ziFijfD7_c9yQCNykwj-L3lwrLks5sD8BOgaJpZM4N1o_U>
> .
>

patdunlavey commented:
Hi David, I don't think I want to merge this until the db creator and db user roles are done. That's an important regression.

dwk2 committed:
two database account roles improves security

two database account roles (db_creator and drupal_user) enables more secure separation of privileges

dwk2 committed:
Security improvement and author correction

two database account users help to better separate privileges; also updated author information

dwk2 commented:
Pat, I've completed the updates that separate the user roles and removes the regression. Please take another look and either merge or let me know if anything else remains. I've tested this code, and it works.

patdunlavey commented:
README.md:
The documentation for creating the db_creator user indicates that the user's name is "db_creator", which only happens to be correct for the sample alias file provided. Also, in markdown, the \` character renders like a `<code>` tag. I think you can escape it with a backslash, like this <code>\\`</code>.

patdunlavey commented:
Or better, just wrap the whole mysql command in `<code></code>` tags.

dwk2 committed:
minor clarifications

minor clarifications and text prettiness.

dwk2 commented:
Edits completed. Please merge.

patdunlavey commented:
Looks good. Thanks for your work on this David, and your patience with my picky review process!
